### PR TITLE
Preffered package in edit customer

### PR DESCRIPTION
**Pull request by ronp2001**
The functionality of the pop-up window for prefferred package has now been put inside of the edit customer pop-up

### CODE REVIEW

File: src/app/services/data-storage.service.ts [hunk 1]
```diff
@@ -90,18 +90,6 @@ export class DataStorageService {
     })
   }
 
-  setCustomerPrefferedPackage(CustomerId: string, prefferedPackageId: string){
```
Comment (suchmememanyskill) [inline src/app/services/data-storage.service.ts:93]: How is the preferred package saved now on the server side? Using `params = params.set('prefferedPackageId', this.prefferredPackage);`?

Comment (suchmememanyskill) [reply src/app/services/data-storage.service.ts:93]: If so, you likely want to use dto's for this, instead of a generic object where you set values on

**Review by suchmememanyskill — APPROVED** [review]
Could use some more polish, but LGTM

File: src/app/customers/edit-customer/edit-customer/edit-customer.component.html [hunk 2]
```diff
@@ -1,45 +1,52 @@
...
-                  <div class="button-group">
-                      <button (click)="onSubmit()" class="done-button">✔ Save</button>
-                      <button (click)="close()" class="close-button">❌ Discard</button>
-                  </div>
-               </div>
-       </div>
-   </div>
-</div>
+    <div class="image-section">
+        <img src="assets/Customer.png" style="width: 100px; height: 100px" class="centered-image" />
```
[the remark this answers is not in the document]
Comment (ronp2001) [reply src/app/customers/edit-customer/edit-customer/edit-customer.component.html:3]: good question, leftover from the old way

File: src/app/customers/edit-customer/edit-customer/edit-customer.component.html [hunk 3]
```diff
@@ -1,45 +1,52 @@
...
+                   <label for="email-input" class="label">E-Mail: </label><input type="text" formControlName="email" id="email-input" class="input" />
+                </div>
+            </div>
+        </form>
+        <div class="footer">
+            <div class="package-input">
+                <label for="package-input" class="label">Preffered Package:</label>
+                <mat-form-field class="select"  id="package-input">
+                    <mat-label>Select a package</mat-label>
+                    <mat-select [(ngModel)]="prefferredPackage">
```
[the remark this answers is not in the document]
Comment (ronp2001) [reply src/app/customers/edit-customer/edit-customer/edit-customer.component.html:38]: copied it from the preferred package component

File: src/app/customers/edit-customer/edit-customer/edit-customer.component.ts [hunk 4]
```diff
@@ -50,6 +55,17 @@ export class EditCustomerComponent implements AfterViewInit, OnDestroy, OnInit {
...
+    this.dataStorageService.getPackagesAndLocations();
+    this.populatePackageList();
+    if (this.customer.preferredPackaging?.id) {
+      this.prefferredPackage = this.customer.preferredPackaging.id
+    }
+  }
+
+  populatePackageList(): void {
+    this.subscription = this.dataStorageService.allInventoryData$.subscribe((inventoryData) => {
+      this.packageList = inventoryData.packageList;
```
[the remark this answers is not in the document]
Comment (ronp2001) [reply src/app/customers/edit-customer/edit-customer/edit-customer.component.ts:67]: again stole it from the old implementation

**Review by Draakje10 — APPROVED** [review]

**Pull request merged**